src/Tribe/Front_Page_View.php

```diff
@@ -103,4 +126,130 @@ protected function get_main_events_url() {
 			return trailingslashit( home_url() . '/' . sanitize_title( $events_slug ) );
 		}
 	}
+
+	/**
+	 * Return the $original_value to avoid convert to a positive integer if the $original_value is the same as
+	 * the ID of the virtual page.
+	 *
+	 * @since TBD
+	 *
+	 * @param $value
+	 * @param $option
+	 * @param $original_value
+	 *
+	 * @return mixed
+	 */
+	public function save_page_on_front( $value, $option, $original_value ) {
+
+		$is_front_page_event_archive = $this->is_virtual_page_id( $original_value );
+
+		tribe_update_option( 'front_page_event_archive', $is_front_page_event_archive );
+
+		return $is_front_page_event_archive ? $original_value : $value;
+	}
+
+	/**
+	 * Add "Main Events Page" option to the Customizer's "Homepage Settings" and the reading settings of the admin
+	 *
+	 * @since TBD
+	 *
+	 * @param string $output HTML output for drop down list of pages.
+	 * @param array  $args   The parsed arguments array.
+	 * @param array  $pages  List of WP_Post objects returned by `get_pages()`
+	 *
+	 * @return string
+	 */
+	public function add_events_page_option( $output, $args, $pages ) {
+
+		// Ensures we only modify the Homepage dropdown.
+		$valid_names = array( '_customize-dropdown-pages-page_on_front', 'page_on_front' );
+		if ( ! isset( $args['name'] ) || ! in_array( $args['name'], $valid_names ) ) {
+			return $output;
+		}
+
+		$label = sprintf(
+			esc_html_x( 'Main %s Page', 'Customizer static front page setting', 'the-events-calendar' ),
+			tribe_get_event_label_plural()
+		);
+
+		$selected = $this->is_page_on_front() ? 'selected' : '';
+		$option = '<option class="level-0" value="' . $this->get_virtual_id() . '" ' . $selected . '>' . $label . '</option></select>';
+		return str_replace( '</select>', $option, $output );
+	}
+
+	/**
+	 * Make sure to set the correct values if we need to update old versions using the previous logic.
+	 *
+	 * @since TBD
+	 */
+	public function backwards_compatible() {
+
+		if ( $this->is_page_on_front() ) {
+			return;
+		}
+
+		// If the archive option is false we don't need to update anything
+		if ( $this->has_event_archive_option() && 'post' === get_option( 'show_on_front' ) ) {
+			update_option( 'show_on_front', 'page' );
```

Comment:
Use a `$let` variable to return if we updated or not. To allow this to be tested later.